docs/troubleshooting/authentication/how_to_retrieve_the_authenticated_type_from_firebase_auth_in_flutterflow.md

```diff
@@ -0,0 +1,60 @@
+---
+keywords: ['firebase', 'auth', 'authenticated']
+slug: /how-to-retrieve-the-authenticated-type-from-firebase-auth-in-flutterflow
+title: How to retrieve the authenticated type from Firebase auth in Flutterflow
+---
+# Retrieve the authenticated type from Firebase auth in Flutterflow
+
+Understanding which authentication method a user has used can be useful for several reasons. For example, it can be leveraged for analytics, user support, and to customize the user's experience based on their login method. This method, however, is specific to Firebase Authentication.​
+
+In our Flutter app, we can find out which method a user used to authenticate by leveraging Firebase's `User.providerData` property. Let's take a closer look at how this works in the code:
+
+```
+import 'package:firebase_auth/firebase_auth.dart';
+
+String getUserSignInMethod() {
+  final user = FirebaseAuth.instance.currentUser;
+  String signInMethod;
+
+  for (var info in user!.providerData) {
+    signInMethod = info.providerId;
+  }
+
+  return signInMethod;
+}
+
+```
+
+Here's a breakdown of the code:
+
+    - We first import the FirebaseAuth package which gives us access to Firebase's authentication methods.
+
+    - Next, we define a function `getUserSignInMethod`. This function will return a string indicating the sign-in method the user used.
+
+    - Inside the function, we obtain the current user from FirebaseAuth using `FirebaseAuth.instance.currentUser`.
+
+We then declare a string `signInMethod` that will store the name of the provider used for sign-in.
```

Comment:
Why are the rest not in bullets or numbered bullets? Why did the bullet stop before this?